### PR TITLE
Added Shop+Sensor Implementation

### PR DESCRIPTION
**Pull request by rafaela-eacf**
This pull request also has code written by Logan and Edward.
The Toolbar, Inventory structure, and HUD were done by Logan. 
Edward implemented the NPC dialogue that gets activated when the player opens the store for the first time. 

Now players can go to the store tab and buy sensors! There are currently two sensors available to buy: the tide sensor and the rain sensor. The first tide sensor is free, then is starts at 5,000 and it increases by 2 times its current price for any additional ones. The rain sensor starts at 5,000 and increases by 2 times its current price for any additional ones as well. 
The information option for the sensors displays more information for the user. The rain sensor does not have an accurate information option because I do not have that information yet.
Once bought, sensors can be placed anywhere that is not already occupied (little bug workaround with how sensors are drawn for now). 
Tide sensors will only be active in ocean tiles (showing a red color) and rain sensors will only be active inland (showing a blue color).
Please play around with it and let me know if anything needs fixing!

### CODE REVIEW

**Review by acher0nparth — CHANGES_REQUESTED** [review]
I think this is great work. The NPC talking to me in the store does a LOT for making the game feel more alive and interactive! The 'more information' buttons are great too, letting the player really see how these sensors work. Currently the rain gauge is using the same diagram as the tide sensor, but it seems like this is intended. 

I think a good feature to add _later_ is if the player tries to place a sensor that they don't have in their inventory, a popup appears asking if they'd like to go to the shop and buy one. 

The only real issues I have are things that I think go back to Logan's PR, #10, mostly about the free stuff in the inventory. If we are going to give free stuff they shouldn't start with anywhere near 20k in cash; they should have enough to cover maybe a month of upkeep costs or something for all the free items. 

I do have one feature I want to see added to this PR directly: the presence of a sensor should be displayed on the bottom bar HUD, just like tiles with structures display their population numbers and stuff down there. The colored dots are good indicators, but when we change to allowing sensors to be placed alongside structures it would be good to have that information elsewhere. Add that feature and I will happily approve this PR.

**Review by lolxela — APPROVED** [review]
Looks really good! I second Blake's request for a popup when there isn't a gauge in inventory because I can see a user getting lost about what to do with those buttons since nothing happens when they try to place them. A NIT comment I have is that the "I'm not sure." button should be replaced with a "No, I need more time" or "Not yet" button or something like that, because it doesn't really seem like an exit button to me and could be interpreted the same as the "ask for help" button. Maybe in the future hovering over an item that's not in inventory can quick display the price of that item?

**Review by devp4 — APPROVED** [review]
Looks great, everything ran great on my end! 

I think it could be helpful to have a popup saying "A sensor is already active on this tile" (or something like that) when placing a sensor on a tile that already has one. Currently, it lets you click "Yes, place sensor" but nothing happens.

File: assets/PLAYER.JSON
Comment (lr2120) [inline assets/PLAYER.JSON:1]: I noticed that there is no punctuation at the end of these sentences (e.g., 'I want to see my achievements'), while other dialogues have punctuation (e.g., 'Welcome to city hall!'). Is this difference in punctuation intentional?

**Comment by lr2120** [comment]
It's working without isse. I had a laugh when I clicked Ask for help and got the response, "The professor recommends...".

One small note, I found a typo in the store dialogue. 
"These sensors wil help you monitor the city and the environment."
It says 'wil' instead of 'will'.

**Review by lr2120 — APPROVED** [review]

**Review by Lchenicek — APPROVED** [review]
Your changes look good to me! I was’t able to get anything to break. Looking forward to all the sensors being implemented.

**Review by acher0nparth — APPROVED** [review]
Awesome! Looks like all my changes were fixed. I bet the popup for lack of sensors can also be used to address my other comment in #10 about adding a popup for lack of funds with minimal work, so that's nice to see as well. Seconding punctuation fixes, but those can just be done as a tack-on to another PR. Great work.

**Review by emenello — APPROVED** [review]
Sorry this is so late. The Shop and Sensors definitely add a whole new element of the game. I know I am going to be working on getting the Ask an Expert connected, but that is for a later PR. For now, this PR showcases a lot of what we worked on last semester and looks polished!

**Comment by acher0nparth** [comment]
Just resolved the merge conflicts between this branch and main. I think I did it right, but I'm not 100% on that. I'll ask in today's meeting (2/14) for others to verify. 
UPDATE [5 minutes later] I did not, in fact, do it right.

**Pull request merged**